packages/shared/src/components/RichTextEditor/plugins/shared/LinkSelector/LinkSelector.tsx

```diff
@@ -0,0 +1,72 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { AppBridgeBlock } from '@frontify/app-bridge';
+import { Button, ButtonEmphasis, ButtonSize, ButtonStyle, ButtonType, IconLink, Modal } from '@frontify/fondue';
+import { useOverlayTriggerState } from '@react-stately/overlays';
+import React, { ReactElement, useEffect, useState } from 'react';
+import { DocumentLinks } from './DocumentLinks';
+
+type LinkSelectorProps = {
+    appBridge: AppBridgeBlock;
+    url: string;
+    onUrlChange: (value: string) => void;
+};
+
+export const LinkSelector = ({ appBridge, url, onUrlChange }: LinkSelectorProps): ReactElement => {
+    const { open: openLinkTree, isOpen: isLinkTreeOpen, close: closeLinkTree } = useOverlayTriggerState({});
+    const [selectedUrl, setSelectedUrl] = useState<string>(url);
+
+    const onSelectUrl = (url: string) => {
+        setSelectedUrl(url);
+    };
+
+    useEffect(() => {
+        if (url && !selectedUrl) {
+            setSelectedUrl(url);
+        }
+    }, [url, selectedUrl]);
+
+    return (
+        <div data-test-id="internal-link-selector">
+            <Button
+                icon={<IconLink />}
+                size={ButtonSize.Medium}
+                type={ButtonType.Button}
+                style={ButtonStyle.Default}
+                emphasis={ButtonEmphasis.Default}
+                onClick={() => openLinkTree()}
+            >
+                Internal link
+            </Button>
+            <Modal zIndex={200} onClose={() => closeLinkTree()} isOpen={isLinkTreeOpen} isDismissable>
```

Comment:
On the Link & Button Flyout the Settings Button is over the Flyout. Is that according to this z-index?

<img width="534" alt="image" src="https://user-images.githubusercontent.com/42832501/231153482-83679db4-d45a-4f4e-a5af-c6f6c395b59c.png">

Reply:
Same issue with all other modals inside the RTE, but it seems to me it's an edge case to have it overlapping. I would not tackled it in this PR.

Reply:
Yes, as the issue was already before I think this is reasonable. Just thought maybe you know a a quick fix as the Link Chooser flyout doesn't have this issue